spec/controllers/text_components_controller_spec.rb

```diff
@@ -91,8 +91,13 @@
       expect(assigns(:text_component).draft?).to be_truthy
     end
 
+    it "removes all associated triggers" do
+      expect(assigns(:text_component).triggers.count == 0)
+    end
+
     it "duplicates questions and answers as well" do
       text_component.question_answers.each.with_index do |qa, i|
+        expect(assigns(:text_component).question_answers[i]).not_to eq(qa)
```

Comment:
I'm pretty sure, this will catch nothing. Ruby has four equality matchers (`==`, `===`, `equal?`, and `eql?`), none of which compares `Active::Record` in a way you would expect here. I would compare the `id` here. That's an integer that must be different.

Reply:
According to the docs it should work: http://api.rubyonrails.org/classes/ActiveRecord/Core.html#method-i-3D-3D

Reply:
Did you try it one time? When I wrote tests like that and I introduced a bug, the rspec never failed. Maybe I can learn sth. here, because I was using the wrong methods.

Reply:
I tried this, but only in the rails console. Will have a closer look tomorrow.

```
irb(main):013:0> tc1 = TextComponent.first
=> #<TextComponent id: 2, …>

irb(main):014:0> tc2 = tc1.dup
=> #<TextComponent id: nil, …>

irb(main):015:0> tc1 == tc1
=> true

irb(main):016:0> tc1 === tc1
=> true

irb(main):017:0> tc1 == tc2
=> false

irb(main):018:0> tc1 === tc2
=> false
```